ext/har/logger.go

```diff
@@ -0,0 +1,110 @@
+package har
+
+
+import (
+    "encoding/json"
+    "net/http"
+    "os"
+    "sync"
+    "time"
+
+    "github.com/elazarl/goproxy"
+)
+
+// Logger implements a HAR logging extension for goproxy
+type Logger struct {
+	mu             sync.Mutex
+	har            *Har
+	captureContent bool
+}
+
+// NewLogger creates a new HAR logger instance
+func NewLogger() *Logger {
+	return &Logger{
+		har: New(),
+	}
+}
+
+// OnRequest handles incoming HTTP requests
+func (l *Logger) OnRequest(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	// Store the start time in context for later use
+	if ctx != nil {
+		ctx.UserData = time.Now()
+	}
+	return req, nil
+}
+
+// OnResponse handles HTTP responses
+func (l *Logger) OnResponse(resp *http.Response, ctx *goproxy.ProxyCtx) *http.Response {
+	if resp == nil || ctx == nil || ctx.Req == nil || ctx.UserData == nil {
+		return resp
+	}
+
+	startTime, ok := ctx.UserData.(time.Time)
+	if !ok {
+		return resp
+	}
+
+	// Create HAR entry
+	entry := Entry{
+		StartedDateTime: startTime,
+		Time:           time.Since(startTime).Milliseconds(),
+		Request:        ParseRequest(ctx.Req, l.captureContent),
+		Response:       ParseResponse(resp, l.captureContent),
+		Cache:          Cache{},
+		Timings: Timings{
+			Send:    0,
+			Wait:    time.Since(startTime).Milliseconds(),
+			Receive: 0,
+		},
+	}
+
+	// Add server IP
+	entry.FillIPAddress(ctx.Req)
+
+	// Add to HAR log thread-safely
+	l.mu.Lock()
+	l.har.AppendEntry(entry)
+	l.mu.Unlock()
+
+	return resp
+}
+
+// SetCaptureContent enables or disables request/response body capture
+func (l *Logger) SetCaptureContent(capture bool) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	l.captureContent = capture
+}
```

Comment:
This functionality complicates the code, just remove the possibility to enable/disable it and keep it always running

Reply:
@ErikPelli I'm going to assume that you don't want the option to disable it and I should remove the function and have it true on initialization.

Reply:
You should completely remove the variable and assume it's always true, yes